### PR TITLE
sound: created a final demo with text to speech

### LINKED ISSUE #1437: sound: create final demo example of text to speech library
Summary: In this issue, we will be creating an example of what text to speech would look like in the chiventure game. This will be done by taking the text to speech libraries we have found, and creating an example using those data structures, and leaving room to build upon it for future teams. We are thinking this could be done by creating an example game in chiventure and using the text to speech functions to create sound through the computer speakers of the text on the in-game screen. Perhaps in the future, it would be worth generalizing this process into specific functions in order to increase ease of implementation.

assignees: Everyone

### PR DESCRIPTION
**Pull request by sangerr**
This pull requests contains a final demo in tts_dialogue_example.c that demonstrates the text to speech feature we have been working on all quarter. This is built upon the previous example we created that contained a dialogue between an npc and character in a game. Based off the tts structs that were created, we added those into the portions of the conversation where the program was printing the responses from the npc. In this game, we have both printed and vocal responses using text to speech.

### CODE REVIEW

**Comment by sangerr** [comment]
There were some issues with the test_sound file I had in my repository, however those were solved by the commits "fixed test sound issue" and "fixed test_sound". To see the actual changes made for the final demo, please look at the "added text to speech to dialogue demo" commit.

**Comment by jwschroeder21** [comment]
Update: we are getting an error that doesn’t allow for the text to speech to run beyond the first line of text. The message popping up in response is “error: entity exists,” and we are further investigating. It is probably the error message of some other module.

**Review by parkerbroz — APPROVED** [review]
Great job! I think this is a promising proof of concept for the application of text-to-speech and the utility of sound in chiventure overall.

File: include/sound/sound.h [hunk 1]
```diff
@@ -43,7 +43,11 @@ typedef union {
  * Parameters:
  *      type: enum SoundType
  *      name: name of the sound, 
+<<<<<<< HEAD
```
Comment (ecoble) [inline include/sound/sound.h]: Remove line 46 and 48, thesis is text created from the merge conflict

File: src/sound/examples/Makefile [hunk 2]
```diff
@@ -1,18 +1,208 @@
-CC=gcc
```
Comment (ecoble) [inline src/sound/examples/Makefile:1]: This file is auto generated by CMake, please remove it

File: src/sound/examples/examples/Makefile [hunk 3]
```diff
@@ -0,0 +1,218 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (ecoble) [inline src/sound/examples/examples/Makefile]: Please remove this as well

**Review by ecoble — CHANGES_REQUESTED** [review]

**Review by ecoble — CHANGES_REQUESTED** [review]
Please also remove all the compiled binary executables, otherwise looks good

**Review by ecoble — APPROVED** [review]

**Comment by sangerr** [comment]
Closing Summary: In this PR we developed a demo that demonstrates the text to speech feature we have been working on all quarter. We did this by adding on to the dialogue between an npc and player using the tts structs to be able to have audio for the conversation. We initially tried to integrate it directly into a game, but had to monkey patch it due to some errors. In addition, we have to restructure the from the example without tts because of errors in printing to cli and speaking in the actual game. Lastly, we had to edit the cmake to make sure that the file would compile and run. We are looking forward to presenting this at the demo on Wednesday!

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Nice job with the demo! I know there were a lot of difficult issues to fix and you did a great job debugging them. Ideally the demo would be debugged before the pull request was created, but otherwise excellent work!

**Pull request merged**